fider: init at 0.24.0

niklaskorz opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

Continuing the previous PR #313713 by @drupol.
Closes #313705.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

drupol committed:
fider: init at 0.24.0

Co-authored-by: Niklas Korz <niklas@niklaskorz.de>

drupol committed:
nixos/fider: init

Co-authored-by: Niklas Korz <niklas@niklaskorz.de>

niklaskorz commented:
@drupol Package and module are functioning now, I'm running a test instance at https://fider.korz.dev/

Do you still want to co-maintain the Fider package and module or shall I remove you from the `maintainers` lists?

drupol commented:
Excellent work, thanks for taking it over, I wonder what was the issue with my initial PR !

You can leave me co-maintainer, I don't mind.

niklaskorz commented:
> Excellent work, thanks for taking it over, I wonder what was the issue with my initial PR !

The blocking issue you described in https://github.com/getfider/fider/issues/1168 I was able to fix by performin the copy of etc in the installPhase and then patching out the runtime copy (see `0001-disable-etc-copy.patch`).

> You can leave me co-maintainer, I don't mind.

Great, thanks :)

Summary of changes from your PR:

- refactored package to allow easy overrides of version, server, frontend, ...
- fixed the etc copy issue
- added example config to NixOS module options
- added release notes

I think this is ready to go for the initial release. I'll add settings for configuring reverse proxies and for type-checked settings (instead of having to specify everything in `environment` / `environmentFile`) in a follow-up PR.

niklaskorz commented:
(running nixpkgs-review for package builds and NixOS tests atm, will post results when done)

drupol commented on pkgs/by-name/fi/fider/server.nix:
```diff
@@ -0,0 +1,30 @@
+  pname = "${pname}-server";
+
+  patches = [
+    ./0001-disable-etc-copy.patch
```
Maybe you should add a comment explaining why the patch is needed, plus a link to the issue.

niklaskorz replied:
Actually let me have another look at the copy mechanism, I went through the code again just now and wonder if there is a better solution than disabling it...

drupol replied:
OK I’ll follow it.

niklaskorz commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 353346`

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fider</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fider</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>fider</li>
  </ul>
</details>
